### PR TITLE
Genre borrower

### PR DESCRIPTION
**Pull request by bmasles**
Finished up the sorting, validation, hidden delete button, and pagination.

### CODE REVIEW

File: src/components/Borrowers.js [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+export class Borrowers extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            first: 0,
+            displayDialog: false,
+            valid: false,
+            nameValid: true,
+            phoneValid: true
+        }
```
Comment (allistergg) [inline src/components/Borrowers.js:17]: You can remove having to bind "this" by using arrow functions on your methods

File: src/store/reducers/borrowersReducer.js [hunk 2]
```diff
@@ -1,5 +1,71 @@
...
+        case READ_BORROWERS_SUCCESFUL:
+            return { ...state, paging: action.data.pop(), borrowers: action.data, loading: false }
+        case READ_BORROWERS_PENDING:
+            return { ...state, loading: true }
+        case READ_BORROWERS_FAILURE:
+            return { ...state, loading: false, error: action.error }
+
+        case CREATE_BORROWERS_SUCCESFUL:
+            newGenres.push(action.data);
+            return { ...state, borrowers: newGenres, loading: false }
```
Comment (allistergg) [inline src/store/reducers/borrowersReducer.js]: I think this should may be new Borrowers

File: src/store/reducers/borrowersReducer.js [hunk 3]
```diff
@@ -1,5 +1,71 @@
...
+    switch (action.type) {
+        case READ_BORROWERS_SUCCESFUL:
+            return { ...state, paging: action.data.pop(), borrowers: action.data, loading: false }
+        case READ_BORROWERS_PENDING:
+            return { ...state, loading: true }
+        case READ_BORROWERS_FAILURE:
+            return { ...state, loading: false, error: action.error }
+
+        case CREATE_BORROWERS_SUCCESFUL:
+            newGenres.push(action.data);
```
Comment (allistergg) [inline src/store/reducers/borrowersReducer.js]: newBorrowers?

File: src/store/reducers/borrowersReducer.js [hunk 4]
```diff
@@ -1,5 +1,71 @@
...
+            return { ...state, borrowers: newGenres, loading: false }
+        case CREATE_BORROWERS_PENDING:
+            return { ...state, loading: true }
+        case CREATE_BORROWERS_FAILURE:
+            return { ...state, loading: false, error: action.error }
+
+        case UPDATE_BORROWERS_SUCCESFUL:
+            //newGenres[newGenres.findIndex(borrower => action.data._id === borrower._id)] = action.data;
+            return { ...state, borrowers: newGenres, loading: false }
```
Comment (allistergg) [inline src/store/reducers/borrowersReducer.js]: newBorrowers?

File: src/store/reducers/borrowersReducer.js [hunk 5]
```diff
@@ -1,5 +1,71 @@
...
+        case UPDATE_BORROWERS_PENDING:
+            return { ...state, loading: true }
+        case UPDATE_BORROWERS_FAILURE:
+            return { ...state, loading: false, error: action.error }
+
+        case DELETE_BORROWERS_SUCCESFUL:
+            // let index = newGenres.findIndex(borrower => action.data._id === borrower._id);
+            // newGenres = newGenres.filter((_, i) => i !== index);
+            return { ...state, borrowers: newGenres, loading: false }
```
Comment (allistergg) [inline src/store/reducers/borrowersReducer.js]: newBorrowers?

**Review by allistergg — CHANGES_REQUESTED** [review]

File: src/components/Borrowers.js (hunk 1 above)
Comment (bmasles) [reply src/components/Borrowers.js:17]: I knew that but forgot it when I was coding for some reason, fixed now

File: src/store/reducers/borrowersReducer.js (hunk 2 above)
Comment (bmasles) [reply src/store/reducers/borrowersReducer.js]: fixed

File: src/store/reducers/borrowersReducer.js (hunk 3 above)
Comment (bmasles) [reply src/store/reducers/borrowersReducer.js]: fixed

File: src/store/reducers/borrowersReducer.js (hunk 4 above)
Comment (bmasles) [reply src/store/reducers/borrowersReducer.js]: fixed

File: src/store/reducers/borrowersReducer.js (hunk 5 above)
Comment (bmasles) [reply src/store/reducers/borrowersReducer.js]: fixed

File: src/store/actions/genreActions.js [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+        error
+    }
+}
+
+const _updateGenresSuccess = (res) => {
+    return {
+        type: UPDATE_GENRES_SUCCESFUL,
+        data: res.data
+    };
+}
```
Comment (sbhuju61) [inline src/store/actions/genreActions.js:140]:
```suggestion
const _updateGenresSuccess = (res) => {
    return {
        type: UPDATE_GENRES_SUCCESFUL
    };
}
```

File: src/store/reducers/borrowersReducer.js [hunk 7]
```diff
@@ -1,5 +1,68 @@
...
+        case CREATE_BORROWERS_SUCCESFUL:
+            newBorrowers.push(action.data);
+            return { ...state, borrowers: newBorrowers, loading: false }
+        case CREATE_BORROWERS_PENDING:
+            return { ...state, loading: true }
+        case CREATE_BORROWERS_FAILURE:
+            return { ...state, loading: false, error: action.error }
+
+        case UPDATE_BORROWERS_SUCCESFUL:
+            return { ...state, borrowers: newBorrowers, loading: false }
```
Comment (sbhuju61) [inline src/store/reducers/borrowersReducer.js:48]:
```suggestion
        case UPDATE_BORROWERS_SUCCESFUL:
            return { ...state, loading: false }
```

File: src/store/actions/genreActions.js (hunk 6 above)
Comment (sbhuju61) [reply src/store/actions/genreActions.js:140]: update and delete request sends empty string from server. `res.data` is empty string

**Review by Jindanupajit-SS — DISMISSED** [review]

**Review by sbhuju61 — DISMISSED** [review]

**Review by allistergg — DISMISSED** [review]

**Review by Jindanupajit-SS — APPROVED** [review]

**Review by sbhuju61 — APPROVED** [review]

**Pull request merged**